File: Codes/3096-minimum-levels-to-gain-more-points.cpp
/*
 * @Author: LetMeFly
 * @Date: 2024-07-19 16:04:29
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-07-19 16:07:48
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
public:
    int minimumLevels(vector<int>& possible) {  // 0减分1加分
        int cnt = 0;
        for (int t : possible) {
            cnt += t == 0 ? -1 : 1;
        }
        int now = 0;
        for (int i = 0; i < possible.size() - 1; i++) {
            now += possible[i] == 0 ? -1 : 1;
            if (now > cnt - now) {
                return i + 1;
            }
        }
        return -1;
    }
};

File: Codes/3096-minimum-levels-to-gain-more-points.go
/*
 * @Author: LetMeFly
 * @Date: 2024-07-19 16:13:21
 * @LastEditors: LetMeFly
 * @LastEditTime: 2024-07-19 16:14:05
 */
package main

func addScore(x int) int {
    if x == 0 {
        return -1
    }
    return 1
}

func minimumLevels(possible []int) int {
    cnt := 0
    for _, t := range possible {
        cnt += addScore(t)
    }
    now := 0
    for i := 0; i < len(possible) - 1; i++ {
        now += addScore(possible[i])
        if now > cnt - now {
            return i + 1
        }
    }
    return -1
}